src/Assessment/Processor/ProcessDatasets.cs

```diff
@@ -2202,47 +2204,97 @@ private bool Create_VM_Opportunity_AsOnPrem_Model(List<VM_Opportunity_AsOnPrem>
             Business_Case_Data.OnPremisesPaaSCost.NetworkCost = BusinessCaseData.OnPremPaaSCostDetails.NetworkCost;
             Business_Case_Data.OnPremisesPaaSCost.SecurityCost = BusinessCaseData.OnPremPaaSCostDetails.SecurityCost;
             Business_Case_Data.OnPremisesPaaSCost.ITStaffCost = BusinessCaseData.OnPremPaaSCostDetails.ITStaffCost;
             Business_Case_Data.OnPremisesPaaSCost.FacilitiesCost = BusinessCaseData.OnPremPaaSCostDetails.FacilitiesCost;
 
-            Business_Case_Data.TotalOnPremisesCost.ComputeLicenseCost = Business_Case_Data.OnPremisesIaaSCost.ComputeLicenseCost + Business_Case_Data.OnPremisesPaaSCost.ComputeLicenseCost;
-            Business_Case_Data.TotalOnPremisesCost.EsuLicenseCost = Business_Case_Data.OnPremisesIaaSCost.EsuLicenseCost + Business_Case_Data.OnPremisesPaaSCost.EsuLicenseCost;
-            Business_Case_Data.TotalOnPremisesCost.StorageCost = Business_Case_Data.OnPremisesIaaSCost.StorageCost + Business_Case_Data.OnPremisesPaaSCost.StorageCost;
-            Business_Case_Data.TotalOnPremisesCost.NetworkCost = Business_Case_Data.OnPremisesIaaSCost.NetworkCost + Business_Case_Data.OnPremisesPaaSCost.NetworkCost;
-            Business_Case_Data.TotalOnPremisesCost.SecurityCost = Business_Case_Data.OnPremisesIaaSCost.SecurityCost + Business_Case_Data.OnPremisesPaaSCost.SecurityCost;
-            Business_Case_Data.TotalOnPremisesCost.ITStaffCost = Business_Case_Data.OnPremisesIaaSCost.ITStaffCost + Business_Case_Data.OnPremisesPaaSCost.ITStaffCost;
-            Business_Case_Data.TotalOnPremisesCost.FacilitiesCost = Business_Case_Data.OnPremisesIaaSCost.FacilitiesCost + Business_Case_Data.OnPremisesPaaSCost.FacilitiesCost;
+            Business_Case_Data.OnPremisesAvsCost.ComputeLicenseCost = 
+                BusinessCaseData.OnPremAvsCostDetails.ComputeLicenseCost - BusinessCaseData.OnPremAvsCostDetails.EsuLicenseCost;
+            Business_Case_Data.OnPremisesAvsCost.EsuLicenseCost = BusinessCaseData.OnPremAvsCostDetails.EsuLicenseCost;
+            Business_Case_Data.OnPremisesAvsCost.StorageCost = BusinessCaseData.OnPremAvsCostDetails.StorageCost;
+            Business_Case_Data.OnPremisesAvsCost.NetworkCost = BusinessCaseData.OnPremAvsCostDetails.NetworkCost;
+            Business_Case_Data.OnPremisesAvsCost.SecurityCost = BusinessCaseData.OnPremAvsCostDetails.SecurityCost;
+            Business_Case_Data.OnPremisesAvsCost.ITStaffCost = BusinessCaseData.OnPremAvsCostDetails.ITStaffCost;
+            Business_Case_Data.OnPremisesAvsCost.FacilitiesCost = BusinessCaseData.OnPremAvsCostDetails.FacilitiesCost;
+
+            Business_Case_Data.TotalOnPremisesCost.ComputeLicenseCost =
+                Business_Case_Data.OnPremisesIaaSCost.ComputeLicenseCost +
+                Business_Case_Data.OnPremisesPaaSCost.ComputeLicenseCost +
+                Business_Case_Data.OnPremisesAvsCost.ComputeLicenseCost;

```

Comment:
Is this validated?
No Null reference error in case of non-avs business case.

Reply:
yes, this has been vaildated. Default values are 0.